Check for extensions in get*Parameter calls

lexaknyazev opened the pull request:
#### bc8dca88d7dbfa281c5098928f205ead75d6d02e
<pre>
Check for extensions in get*Parameter calls
<a href="https://bugs.webkit.org/show_bug.cgi?id=223347">https://bugs.webkit.org/show_bug.cgi?id=223347</a>

Reviewed by Kenneth Russell.

When EXT_texture_filter_anisotropic extension is enabled,
getSamplerParameter may be used to query sampler&apos;s max anisotropy.

Although ANGLE generates INVALID_ENUM when extensions are not enabled,
it cannot return null as required by the WebGL spec, so WebKit has to
check for them anyway.

See
* conformance2/extensions/ext-texture-filter-anisotropic.html
* conformance2/extensions/oes-draw-buffers-indexed.html

* LayoutTests/TestExpectations:
* Source/WebCore/html/canvas/WebGL2RenderingContext.cpp:
(WebCore::WebGL2RenderingContext::getIndexedParameter):
(WebCore::WebGL2RenderingContext::getSamplerParameter):

Canonical link: <a href="https://commits.webkit.org/252895@main">https://commits.webkit.org/252895@main</a>
</pre>

kenrussell reviewed (approved):
Looks good overall.

Can you enable any new conformance tests under webgl/2.0.y/ in TestExpectations along with this code change?

lexaknyazev committed:
Check for extensions in get*Parameter calls
https://bugs.webkit.org/show_bug.cgi?id=223347

Reviewed by Kenneth Russell.

When EXT_texture_filter_anisotropic extension is enabled,
getSamplerParameter may be used to query sampler's max anisotropy.

Although ANGLE generates INVALID_ENUM when extensions are not enabled,
it cannot return null as required by the WebGL spec, so WebKit has to
check for them anyway.

See
* conformance2/extensions/ext-texture-filter-anisotropic.html
* conformance2/extensions/oes-draw-buffers-indexed.html

* LayoutTests/TestExpectations:
* Source/WebCore/html/canvas/WebGL2RenderingContext.cpp:
(WebCore::WebGL2RenderingContext::getIndexedParameter):
(WebCore::WebGL2RenderingContext::getSamplerParameter):

Canonical link: https://commits.webkit.org/252895@main
